### ISSUE TITLE
[Proposal]: Add Http proxy support.

### ISSUE THREAD

**Issue by mike820324** [mike820324/microProxy#202]
Adding HTTP Proxy Support.
There may be two cases when firefox is configured as http proxy,
### Firefox Behavior:
- Same Protocol (HTTP Proxy to HTTP site):
  The query message would be something like the following message. 

```
GET http://www.google.com HTTP/1.1\r\n
...
```
- Different Protocol(HTTP Proxy to HTTPs site):
  Firefox will use CONNECT method to deal with it.

```
CONNECT https://www.google.com HTTP/1.1\r\n
User-Agent: xxx
Proxy-Connection: keep-alive
Connection: keep-alive
Host: www.google.com:443\r\n\r\n
```

The following curl command serve the same functionality.

```
curl --proxy http://127.0.0.1:5580 http://www.google.com
curl --proxy http://127.0.0.1:5580 https://www.google.com
```

P.S
curl also has the proxytunnel option, which will force all the query to use connect instead.

```
curl --proxy http://127.0.0.1:5580 --proxytunnel http://www.google.com
```

**Comment by chhsiao90** [comment]
> ```
> curl --proxy http://127.0.0.1:5580 http://www.google.com
> curl --proxy http://127.0.0.1:5580 https://www.google.com
> ```

Maybe you mean
```
curl --proxy http://127.0.0.1:5580 https://www.google.com
curl --proxy http://127.0.0.1:5580 --proxytunnel https://www.google.com
```

**Comment by mike820324** [comment]
Yes, the proxytunnel options will force curl to use CONNECT in **every** protocol. 

The above comment that I want to describe is how http client such as curl and firefox handle http proxy in different situations. :)

So if one want to check the CONNECT method, either

- using http proxy connect to https website

or

- use proxytunnel instead.

If one want to check the other way,

- using http proxy connect to http website should create this issue.

Sorry for the misunderstanding.

**Comment by mike820324** [comment]
Closed via #235
